Assets/Game/Scripts/DebugControl.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace PK.InfiniteRunner.Debuging
+{
+    public enum DebugType
+    {
+        Normal,
+        Warning,
+        Error
+    }
+    public class DebugControl : Singelton<DebugControl>
+    {
+        [Header("Ui")]
+        [SerializeField]
+        private Text displayText;
+
+        [SerializeField]
+        private Image normalBotton;
+        [SerializeField]
+        private Image warningBotton;
+        [SerializeField]
+        private Image errorBotton;
+
+        [SerializeField]
+        private Color enableColor;
+        [SerializeField]
+        private Color disableColor;
+
+        private List<DebugText> textList = new List<DebugText>();
+
+        private bool showNormalText = true;
+        private bool showWarningText = true;
+        private bool showErrorText = true;
+
```

Comment:
OnAwake()
{
//IF the debugUI prefab does not exist in the scene( the UI). Add it to the scene programmatically using Instantiate. This way any scene in the game can have access to the debugUI, by just calling the singleton instance. 



Also bind also the reference to each button,warning button, error button here. Simply findGameobject by name.


}

 private void DisplayTextOfType(DebugText nText)  ...following coding conventions. it should be like this
DisplayTextOfType(DebugText debugText)


Lastly do a spell check on your code :P  Botton is spelt Button.